### ISSUE TITLE
[dupl] warning: 'SecTrustedApplicationCreateFromPath' is deprecated

### ISSUE THREAD

**Issue by jonioni** [99designs/keyring#66]
Related: #56, #59, #60

## Problem

Encountering the following errors on macOS Catalina 10.15 with the latest keyring release:

```bash
cgo-gcc-prolog:203:11: warning: 'SecTrustedApplicationCreateFromPath' is deprecated: first deprecated in macOS 10.15 - No longer supported [-Wdeprecated-declarations]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Security.framework/Headers/SecTrustedApplication.h:59:10: note: 'SecTrustedApplicationCreateFromPath' has been explicitly marked deprecated here
```

This is reported in [keybase/go-keychain #58](https://github.com/keybase/go-keychain/pull/58) and seems to have been fixed in [keybase/go-keychain #60](https://github.com/keybase/go-keychain/pull/60).

## Proposal

Upgrade the corresponding dependency to use a newer version of keybase/go-keychain.

**Comment by mtibben** [comment]
Temporary fix in db030e0c0975ed003e02a6049e8be23aec8daf9c

**Comment by flowchartsman** [comment]
How is this one looking? Is there any reason it's still pinned to your fork of go-keychain?

**Comment by mtibben** [comment]
@flowchartsman Yes, there is a reason. The removal of `SecTrustedApplicationCreateFromPath` changes the security behaviour of aws-vault. It removes the ability for aws-vault to remove itself from the credential ACL, which is required to get keychain prompts when a credential is accessed. See https://github.com/99designs/keyring/pull/59#issuecomment-584522383 for further discussion

**Comment by mtibben** [comment]
I believe using `SecACLUpdateAuthorizations` is the new way to update ACLs

**Comment by flowchartsman** [comment]
Great, thanks for the explanation! Please let me know if there's any legwork that needs done.

**Comment by mtibben** [comment]
The work required is to remove use of the fork, but find a way to keep the ACL-clearing behaviour
